packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -737,6 +739,35 @@ abstract class ResidentRunner {
     throw Exception('Canvaskit not supported by this runner.');
   }
 
+  /// Write the SkSL shaders to a zip file in build directory.
+  Future<void> writeSkSL() async {
+    if (!supportsWriteSkSL) {
+      throw Exception('writeSkSL is not supported by this runner.');
+    }
+    final Map<String, Object> data = await flutterDevices.first.views.first.getSkSLs();
+    if (data.isEmpty) {
+      globals.logger.printStatus(
+        'No data was receieved. To ensure SkSL data can be generated use a '
+        'physical device then:\n'
+        '  1. Pass "--cache-sksl" as an argument to flutter run.\n'
+        '  2. Interact with the application to force shaders to be compiled.\n'
+      );
+      return;
+    }
+    final File outputFile = globals.fsUtils.getUniqueFile(
+      globals.fs.currentDirectory,
+      'flutter',
+      'sksl',
+    );
+    final Map<String, Object> manifest = <String, Object>{
+      'device': getNameForTargetPlatform(await flutterDevices.first.device.targetPlatform),
```

Comment:
Other than the `targetPlatform`, it might be nice to also record the exact device type, such as whether it's a Moto G4, or a Pixel 4. I'm not sure if our tools already have that information. If not, maybe just write down the device name, i.e., the first column of the following `flutter devices` output:
```
GM1910                 • 370d7342                                 • android-arm64  • Android 10 (API 29)
Moto G 4               • ZY2245H6KS                               • android-arm    • Android 7.0 (API 24)
SixDegreesOfSeparation • 2b398993d86f42b05511b330a89ae65289832852 • ios            • iOS 12.4.1
Chrome                 • chrome                                   • web-javascript • Google Chrome 80.0.3987.149
Web Server             • web-server                               • web-javascript • Flutter Tools
```

Reply:
Done